File: spec/java/src/io/kaitai/struct/specwrite/TestStrPadTermRoundtrip.java
package io.kaitai.struct.specwrite;

import io.kaitai.struct.KaitaiStruct.ReadWrite;
import io.kaitai.struct.testwrite.StrPadTermRoundtrip;
import org.testng.annotations.Test;

public class TestStrPadTermRoundtrip extends CommonSpec {
    @Test
    public void testStrPadTermRoundtrip() throws Exception {
        // NOTE: here it makes sense to prefer a manual test over the automatic
        // testReadWriteRoundtrip, because the roundtrip can't always recognize whether the
        // `pad-right` is correct (i.e. whether the serialized field is padded correctly)

        StrPadTermRoundtrip r = new StrPadTermRoundtrip();

        r.setStrPad("str1");
        r.setStrTerm("str2foo");
        r.setStrTermAndPad("str+++3bar+++");
        r.setStrTermInclude("str4baz@");

        assertEqualToFullFile(r, "str_pad_term.bin");
    }

    @Override
    protected Class<? extends ReadWrite> getStructClass() {
        return StrPadTermRoundtrip.class;
    }

    @Override
    protected String getSrcFilename() {
        return "str_pad_term.bin";
    }
}

File: spec/python/specwrite/test_str_pad_term_roundtrip.py
import unittest
from specwrite.common_spec import CommonSpec

from testwrite.str_pad_term_roundtrip import StrPadTermRoundtrip

class TestStrPadTermRoundtrip(CommonSpec.Base):
    def __init__(self, *args, **kwargs):
        super(TestStrPadTermRoundtrip, self).__init__(*args, **kwargs)
        self.struct_class = StrPadTermRoundtrip
        self.src_filename = 'src/str_pad_term.bin'

    def test_str_pad_term_roundtrip(self):
        # NOTE: here it makes sense to prefer a manual test over the automatic
        # testReadWriteRoundtrip, because the roundtrip can't always recognize whether the
        # `pad-right` is correct (i.e. whether the serialized field is padded correctly)

        r = StrPadTermRoundtrip()

        r.str_pad = "str1"
        r.str_term = "str2foo"
        r.str_term_and_pad = "str+++3bar+++"
        r.str_term_include = "str4baz@"

        self.assert_equal_to_full_file(r, 'src/str_pad_term.bin')
